app/assets/scripts/reducers/region.js

```diff
@@ -0,0 +1,245 @@
+'use strict';
+import { combineReducers } from 'redux';
+import _toNumber from 'lodash.tonumber';
+import _groupBy from 'lodash.groupby';
+
+import { getCentroid } from '../utils/country-centroids';
+import { groupByDisasterType } from '../utils/utils';
+
+const initialState = {
+  fetching: false,
+  fetched: false,
+  receivedAt: null,
+  data: {}
+};
+
+const stateInflight = (state, action) => {
+  return Object.assign({}, state, {
+    error: null,
+    fetching: true,
+    fetched: false
+  });
+};
+
+const stateError = (state, action) => {
+  return Object.assign({}, state, {
+    fetching: false,
+    fetched: true,
+    receivedAt: action.receivedAt,
+    error: action.error
+  });
+};
+
+const stateSuccess = (state, action) => {
+  return Object.assign({}, state, {
+    fetching: false,
+    fetched: true,
+    receivedAt: action.receivedAt,
+    data: action.data
+  });
+};
+
+function regionData (state = initialState, action) {
+  switch (action.type) {
+    case 'GET_REGION_INFLIGHT':
+      state = Object.assign({}, state, {
+        [action.id]: stateInflight(state, action)
+      });
+      state = stateInflight(state, action, action.id);
```

Comment:
What does this second state assignment do?

Reply:
Nothing. that's a mistake